python/feathub/dsl/built_in_func.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright 2022 The FeatHub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from enum import Enum
+from typing import List
+
+from feathub.common.exceptions import FeathubExpressionException
+from feathub.common.types import DType, String, Int64
+
+
+class BuiltInFunc(Enum):
+    LOWER = "LOWER"
+    UNIX_TIMESTAMP = "UNIX_TIMESTAMP"
+
+    @staticmethod
+    def get(func_name: str) -> "BuiltInFunc":
+        return BuiltInFunc(func_name)
+
+    def get_result_type(self, input_types: List[DType]) -> DType:
+        """
+        Get the result data type of the builtin function.
+        """
+        if self == BuiltInFunc.LOWER:
+            return String
+        elif self == BuiltInFunc.UNIX_TIMESTAMP:
```

Comment:
I'm not sure enum would be a good design for built-in functions, since we may have hundreds of functions to support. Would it be better to adopt another design pattern, like `org.apache.flink.table.functions.BuiltInFunctionDefinitions`?